src/inference/src/cpp/ie_infer_request.cpp

```diff
@@ -492,9 +495,11 @@ void InferRequest::set_callback(std::function<void(std::exception_ptr)> callback
 
 std::vector<VariableState> InferRequest::query_state() {
     std::vector<VariableState> variable_states;
+    std::vector<std::shared_ptr<void>> soVec;
     OV_INFER_REQ_CALL_STATEMENT({
+        soVec = {_so, _impl->getPointerToSo()};
```

Comment:
since the `_impl` is created by a `_impl->getPointerToSo()`, do we need to hold a reference to `_so`? From the first glance, it seems useless here. The same for all other places.

Reply:
_so here is from the infer request. _impl->getPointerToSo() is the so from iInferRequest, which is something I add in this PR for lifecycle management for virtual plugins.

Reply:
am I right that `getPointerToSo()` is an additional so from hardware plugin?
1. For case AUTO:GPU so=AUTO, getPointerToSo=GPU
2. for GPU, both so and getPointerToSo are GPU
right?

Reply:
1. yes
2. for GPU, so is GPU, getPointerToSo is nullptr, as GPU plugin does not need to save an extra copy of gpu library.